### PR TITLE
document the return types of `fieldname` and `fieldnames`

### PR DESCRIPTION
**Pull request by nsajko**

### CODE REVIEW

**Comment by nsajko** [comment]
The current PR gives more information than that, though. The return type is documented conditional on the argument. So what you propose would be a loss of information relative to the current PR?

Also I don't like that pattern because it's not Julia syntax.

**Comment by nsajko** [comment]
bump

File: base/reflection.jl [hunk 1]
```diff
@@ -246,6 +251,8 @@ fieldname(t::Type{<:Tuple}, i::Integer) =
 
 Get a tuple with the names of the fields of a `DataType`.
 
+For a tuple type, each name is an `Int` (the index of the field), otherwise it's a `Symbol`.
```
Comment (fingolfin) [inline base/reflection.jl:254]: I would spell out the colloquialism "it's" as "it is".

Perhaps also have the "common" case first? Like this:
```suggestion
The field names are of type `Symbol`, except for tuple types, where they are
of type `Int` (the index of the field).
```

File: base/reflection.jl [hunk 2]
```diff
@@ -212,13 +212,18 @@ end
 
 Get the name of field `i` of a `DataType`.
 
+The return type is `Int` (the index of the field) for `x <: Tuple`, otherwise the return type is `Symbol`.
```
Comment (fingolfin) [inline base/reflection.jl:215]: As for my other remark, I would suggest having the "common case" first.

Anyway, good idea to document this and to add the examples, thanks!

**Review by fingolfin — APPROVED** [review]

**Pull request merged**